lib/plugins/chat.js

```diff
@@ -134,12 +128,13 @@ function inject (bot, options) {
 
       const parameters = {
         sender,
-        content
+        content: message ? JSON.parse(message) : { text: data.plainMessage }
```

Comment:
Handling it this way breaks chat patterns

Reply:
Which chat pattern? Do you mean passing to ChatMessage.fromNetwork ? Shouldn't be a problem if it's already formatted

Reply:
Or are you saying we can skip the call to fromNetwork?

Reply:
If we don't consider the unsigned content then we might miss server formatting

Reply:
Without having tested it, I can say pretty confidently that most non-Vanilla server implementations will send the message as unsigned content.

Reply:
There is another prismarine chat message in the .unsigned field. Can you explain the formatting issue here?

Reply:
msg.toString() will not be the expected message in a lot of cases so `messagestr` event will emit an "unpredictable" message. That means chat patterns added with `addChatPattern` will not necessarily match messages as they appear in the Vanilla Client. And a lot of messages can't possibly be matched because we're only accounting for the raw message

Reply:
If we don't want to parse unsigned contents internally in Mineflayer then we shouldn't parse formattedMessage either

Reply:
Under the previous commit these unsigned messages would not show either unless you set a special option in the bot, so not an optimal fix there.

Reply:
I did remove that check in my latest commit

Reply:
See https://github.com/PrismarineJS/mineflayer/pull/2894/commits/0229bd73ab230d2b6de77058007c94d7f9460f95

Reply:
I still believe the check should be there though, as that would make it easier to simulate the Vanilla client (so you don't have to disable secure chat only in vanilla MC to mimic mineflayer behaviour)

Reply:
But we can always add that check later so let's not take that discussion in this PR.